kani-driver/src/harness_runner.rs

```diff
@@ -139,11 +139,31 @@ impl KaniSession {
                     "Complete - {succeeding} successfully verified harnesses, {failing} failures, {total} total."
                 );
             } else {
-                // TODO: This could use a better error message, possibly with links to Kani documentation.
-                // New users may encounter this and could use a pointer to how to write proof harnesses.
-                println!(
-                    "No proof harnesses (functions with #[kani::proof]) were found to verify."
-                );
+                match (self.args.harnesses.as_slice(), &self.args.function) {
+                    ([], None) =>
+                    // TODO: This could use a better error message, possibly with links to Kani documentation.
+                    // New users may encounter this and could use a pointer to how to write proof harnesses.
+                    {
+                        println!(
+                            "No proof harnesses (functions with #[kani::proof]) were found to verify."
+                        )
+                    }
+                    ([harness], None) => {
+                        println!("No harnesses matched the harness filter: `{harness}`",)
+                    }
+                    (harnesses, None) => {
+                        println!(
+                            "No harnesses matched the harness filters: `{}`",
+                            harnesses.join("`, `")
+                        )
+                    }
```

Comment:
I'm concerned about this scenario:
 * User has Kani proofs running in CI
 * User renames proofs or embed them in a non-visible module for Kani.
 * Kani running in CI doesn't find proofs, or the filters cause it to not run them.
In this scenario, Kani would silently continue to run no proofs until someone takes a look at the CI logs.

Of course, the same scenario can happen while still running some proofs. In that case, there isn't much we can do about. But we can prevent a scenario like the one I described earlier by returning an error here, and an exit code different to 0 in the caller of this function (not sure how this structured, just giving ideas).

I'll admit that this is a very conservative approach, but I think users would appreciate the effort to prevent this problem.

Reply:
BTW I know that we had https://github.com/model-checking/kani/issues/1857 in the past, but I'm just wondering if we should reconsider it.

Reply:
I totally understand the concern. There is a question here about what the level of flexibility should be.

I think your concern is legit, and it is something that I noticed happens sometimes when we add tests in our regression and we forget to add the `expected` files. We think something is being tested, but it's not.

I'm not sure that throwing an error when the filter doesn't apply to any harness really solves the root problem though. Let's say the user filter use to match one harness and it doesn't anymore because that harness got renamed. If that filter also matches another harness, Kani will run without throwing an error. So in this case, we are now missing a harness that we didn't before.

Reply:
Maybe we can report the harnesses that we skipped. On verbose mode, we could even print them. What do you think?

Reply:
I noticed that `cargo tests` include the number of tests filtered out:
```
test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 38 filtered out; finished in 0.00s
```

Reply:
Including the number of filtered-out harnesses sounds like a reasonable solution.

Reply:
Yes, as said earlier: there isn't a lot we can do when other harnesses match the filter. But the case where the user runs `--harness` and filters out all proof harnesses should be signaled somehow.

Reply:
Fair... do you want it to be an error or a warning?

Reply:
An error.